### DISCUSSION TITLE
[Feature request] remember scroll position for each note

### DISCUSSION THREAD

**Discussion by rivol** [standardnotes/forum#2514] [Suggestions]
I often find myself switching between multiple notes, either looking up information that's spread among them, or to move info from one note to another. It would help a lot if each note would keep it's scroll position, independently of other notes.

ATM this seems to be editor specific (tested on Linux desktop app):
- plain editor keeps a global scroll position, ie all notes share a single one.
- minimal markdown editor resets scroll position to the top of the note whenever I open another note.

I don't think it's important to sync scroll positions among devices, so they could just be saved in application's (or editor extension's) state. I (probably) also wouldn't care much if the scroll positions are lost at application restart.

**Comment by atmoio** [comment]
Hmm..this would be somewhat trivial if we were only focusing on the plain editor. But supporting this for every editor might be a little more tricky. I'll keep this in mind.

**Comment by rivol** [comment]
If you could just save/restore the pixel offset of the editor's view (I'm assuming here that each editor is a scrollable area at the topmost level, no idea if that's actually the case...), then that _might_ be a sane default that editors could then further customize if need be.

**Comment by atmoio** [comment]
The thing is each editor would need to implement custom logic for this. Would be a considerable amount of work.

**Comment by erehwonstudios** [comment]
wanted to second this idea, i tend to keep logs chronologically top-down (most natural) so when i'm switching around i find myself repetitively scrolling, scrolling, and scrolling to where i was.  thought perhaps i could try a keyword marker for last position, but search doesn't go to the targets either, just the notes.

since your save granularity is down to the keystroke, would it be possible to retain the line number of the cursor loc when a note is "unviewed" ("minimise" doesn't make sense here?) and optionally jump back to that line number when the note is raised / viewed again?

as 'rivol' said above, doesn't need to be any wider scope than current session.  i know you're super busy and there are a lot of editors, but (perhaps) a simple line_num solution might work on ALL of your editors.  or maybe not.  thanks for your time, and the great work you've done.

**Comment by atmoio** [comment]
@erehwonstudios it might be of moderate difficulty to implement this in the plain editor. But for external editors, we couldn't do something like line num, since every editor would have to implement its own logic (i.e what would be the line num in the simple task editor? It doesn't use regular lines). 

We could probably have a javascript listener for scroll events and save that, but that's just something I would have to investigate. I've made note of investigating this as a backlog item.

**Comment by erehwonstudios** [comment]
i understand it might not work for editors that don't have line numbers, but it sure would be useful for the ones that do.

btw, am i crazy or did you implement a test of this on the plain editor for like an hour yesterday?  because it was definitely remembering positions for a short time, then it reverted to always returning to line 1.  and, if it was you, how would such a mod get pushed to the standalone apps without an "update"?

anyway, thanks for looking at it and the quick response.  much appreciated!

**Comment by atmoio** [comment]
Lol no, no such update was done. Not sure why you would have seen that behavior.

**Comment by bschlagel** [comment]
Just popping back in to say I would love this feature, and strongly suggest making it a high priority UX improvement. Even just for the plain editor to start would be great. It's a very frequent usability issue for me, one of the  rare janky-feeling things about the app in my experience.

Also, may be a different issue but on iPad one thing I'm noticing right now — even editing a single note, if I switch apps + then return to SN, it seems to reset the scroll position…maybe the fault of iOS being over aggressive with memory management or something, but even so, all the more reason to store scroll position b/c it feels incredibly buggy to toggle back to an app I was just using and be in a completely different place in a note, makes it almost unusable for certain things.

**Comment by fompahahaha** [comment]
This feature works very well in Pure Writer, OneNote, Nimbus Note and I like it, it’s inconvenient without it. Notion now does this by pressing "Back". Browsers, instant messengers and even some file managers return to their exact original position when returning to the previous page.

**Comment by JaspalSuri** [comment]
@bschlagel I'm sorry to hear that. I've passed that information along to the team under the feature request. 

@fompahahaha thanks for the feedback, I'll pass along Notion's approach as well.
